helix-core/src/uri.rs

```diff
@@ -0,0 +1,102 @@
+use std::path::{Path, PathBuf};
+
+/// A generic pointer to a file location.
+///
+/// Currently this type only supports paths to local files.
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+#[non_exhaustive]
+pub enum Uri {
+    Path(PathBuf),
+}
+
+impl Uri {
+    pub fn from_file_path<P: AsRef<Path>>(path: P) -> Self {
+        Self::Path(path.as_ref().to_path_buf())
+    }
+
+    // This clippy allow mirrors url::Url::from_file_path
+    #[allow(clippy::result_unit_err)]
+    pub fn to_url(&self) -> Result<url::Url, ()> {
+        match self {
+            Uri::Path(path) => url::Url::from_file_path(path),
+        }
+    }
+}
+
+impl From<Uri> for PathBuf {
+    fn from(uri: Uri) -> Self {
+        match uri {
+            Uri::Path(path) => path,
+        }
+    }
+}
```

Comment:
i think this should be `TryFrom` rather than `From`, especially considering `Uri` is `#[non_exhaustive]`.

Reply:
aha, you caught me being lazy :P